Develop

xcixor opened the pull request:
#### Purpose of this PR
Complete the required api endpoints for challenge 3

#### Tasks completed
* Add functionality to register user
* Add functionality to login user
* Add functionality to create a ride
* Add functionality to get all rides
* Add functionality to get a ride
* Add functionality request a ride
* Add functionality to reject or accept a ride
* Add functionality to view ride's requests
* Protect routes with jwt
#### How should this be manually tested?
* Clone the repo
* Create a virtualenv to install pytest
* Run the command pytest tests/ -v
#### Questions
* How can this project be improved
#### Relevant pivotal tracker stories
[#158697162,   #158696017,   #158695828,   #158695946,   #158696086,   #158696206](https://www.pivotaltracker.com/n/projects/2179559)

xcixor committed:
Merge pull request #10 from xcixor/158549047-get-specific-ride

158549047 get specific ride

xcixor committed:
tests(user functionality)

-Add test for user model and endpoints.

xcixor committed:
tests(rides)

-Add tests for ride endpoints

xcixor committed:
Fix(typo)

-Remove typing errors

xcixor committed:
feat(registration)

-Add functionality to create user table

-Implement endpoint to register user

xcixor committed:
tests(models)

-Add tests for models

xcixor committed:
-Add boiler plate code for creating rides

xcixor committed:
resove conflict

xcixor committed:
Merge branch 'develop' of github.com:xcixor/my-ride into develop

xcixor committed:
fix typo

xcixor committed:
fix(TDD)

-Redo app tests.

xcixor committed:
feat(user registration)

-Add functionality to create user in models and controller

-Create endpoint to register user

xcixor committed:
feat(login): [#158695738])

-Add functionality to verify user password.

xcixor committed:
feat(login)

-Implement endpoint for user login

-Add route for login.

-Implement jwt generation

[Finishes

xcixor committed:
Merge branch '158695738-login' of github.com:xcixor/my-ride into 158695738-login

xcixor committed:
feat(create ride)

-Add database functionality to create a ride

Implement endpoint to create a ride

xcixor committed:
feat(get rides)

-Add fuctionality to query db for rides

-Add endpoint to display the rides

xcixor committed:
feat(get ride)

-Query db for a specific ride

-Add endpoint to display the data

xcixor committed:
feat(make request)

-Add functionality to update db of a new request

-Add endpoint to make request

xcixor committed:
refactor

-Edit tests error messages to conform with return values of functions.

xcixor committed:
feat(get requests)

-Add function to query database for user's ride request

-Add edndpoint to display ride's requests to user

[Delivers #158696086]

xcixor committed:
feat(accept requests)

-Add functionality to update requests table of the drivers decision

-Add endpoint to enable the user to update the request

xcixor committed:
-Add travis script to create db.

xcixor committed:
-Add app dependencies

xcixor committed:
-Add configurations for db in config file

xcixor committed:
-Edit badges to reflect current progress

xcixor committed:
Update travis.yml for online tests running.

xcixor committed:
Add docstring for test method.

JoshuaOndieki commented:
Implement feedback made by Codacy. Also, your CI is not connecting with the DB
According to your Travis logs, the host is being set to none

xcixor committed:
Edit config file to resolve travis conflict

xcixor committed:
Edit classes to initialize with values

Name returned values

JoshuaOndieki commented:
The [deployment](https://dashboard.heroku.com/apps/my-ride/activity/builds/f4eb64f2-f490-44b0-91c8-600b5e641d03) you made is buggy :bug: contrary to your CI status
Have you looked into what could be the problem as the CI is passing

xcixor committed:
Add error message for db delete failure

xcixor committed:
fix(endpoints)

-Return appropriate error for failure scenarios

Return error message if inexisting ride is requested

xcixor committed:
Decrement ride capacity on accept ride.

xcixor committed:
Fix test urls to conform to new routes naming

xcixor committed:
-Edit db in config file to avoid conflict between testdb and development db

xcixor committed:
Add decorator to prevent string ids from being passes

xcixor committed:
improve code quality
